src/Umbraco.Core/Services/ContentService.cs

```diff
@@ -2695,8 +2695,11 @@ public bool RecycleBinSmells()
                 var total = long.MaxValue;
                 while (page * pageSize < total)
                 {
+                    // when copying a branch into itself, the copy of a root would be seen as a descendant
+                    // and would be copied again => filter it out.
                     IEnumerable<IContent> descendants =
-                        GetPagedDescendants(content.Id, page++, pageSize, out total);
+                        GetPagedDescendants(content.Id, page++, pageSize, out total)
+                            .Where(descendant => descendant.Id != copy.Id);
```

Comment:
Why not do the check in the the foreach below. Then we are not iterating the descendants multiple times.

Reply:
I can fix it if you agree